src/XLine.js

```diff
@@ -47,13 +59,21 @@ export default class XLine extends React.Component {
     const className = `rct-chart-line-x ${this.props.className}`;
     const lineX = xScale(value);
 
+    let y1 = -spacingTop;
+    let y2 = height + spacingBottom;
+
+    if (yLimit) {
+      y1 = yScale(yDomain[0]);
```

Comment:
<img width="1135" alt="Screen Shot 2019-06-12 at 11 23 21 PM" src="https://user-images.githubusercontent.com/14301342/59375040-e3d5cc00-8d6a-11e9-87c0-2d577c0300cb.png">

So with y1 set as -spacingTop, the line begins from the top instead of the bottom. I assumed since the value is within the domain, so the line should begin from the lower bound of the domain. Let me know if that's not what you had in mind/if there's a better implementation for this.

Reply:
Gotcha! I think we're missing the spacingBottom as well so we need to account for that if users pass that in. Is it possible, for consistency, for the YLine to always grow from the top or bottom (the latter makes more sense to me). So if there's a yLimit the only thing that would be altered is y2 where y1 should stay `yScale(yDomain[0]) - spacingBottom`?

Reply:
Hey @krissalvador27, sorry for the delay in making these changes. Finally had the chance to get to them. 

For the XLine to always consistently grow from the bottom (ie. by fixing the value of y1 as `yScale(yDomain[0]) - spacingBottom`), I will also have to account for the XLines created by the XGrid, which is a child of the XAxis component. 

As of now, the XAxis component doesn't accept the yDomain as a prop from the XYPlot, and so the XLines that make the grid do not have the prop yDomain available. So should I go ahead and add yDomain and yScale as a prop for both XAxis and XGrid to make all lines (including the grid) lines grow from the bottom?

Reply:
Hey @tanayv!

Thanks for catching that! XYPlot should hand yDomain and yScale down to all its children, so XAxis should easily accept it. I do worry that the complexity and scope is growing though. I think we can go with your original logic to accomplish the original XLine and YLine height and width limits and consistently have XLines grow from the top/limit point. 👍